### DISCUSSION TITLE
How to change fileMatch for Docker manager

### DISCUSSION THREAD

**Discussion by fgreinacher** [renovatebot/renovate#31688] [Request Help] [answered]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Mend Renovate hosted app on github.com

### Please tell us more about your question or problem

I'm trying to change the fileMatch for the Docker manager as suggested in https://github.com/renovatebot/renovate/issues/15975#issuecomment-1214640697.

My Renovate config looks like this:

```json
{
  "extends": ["config:base"],
  "docker": {
    "fileMatch": [
      "(^|/)Earthfile$",
      "(^|/|\\.)Dockerfile$",
      "(^|/)Dockerfile[^/]*$"
    ]
  }
}
```

But Renovate logs a warning and uses the default `fileMatch`:

```
WARN: Found renovate config warnings
{
  "warnings": [
    {
      "topic": "Config warning",
      "message": "\"fileMatch\" must be configured in a manager block and not here: packageRules[0]"
    }
  ]
}
```

The warning message left me a bit confused because it mentions a package rule even though I don't have any.

I then noticed that my config is being migrated:

```
DEBUG: Config migration necessary
{
  "oldConfig": {
    "extends": [
      "mergeConfidence:all-badges",
      "config:base"
    ],
    "docker": {
      "fileMatch": [
        "(^|/)Earthfile$",
        "(^|/|\\.)Dockerfile$",
        "(^|/)Dockerfile[^/]*$"
      ]
    }
  }
  "newConfig": {
    "extends": [
      "mergeConfidence:all-badges",
      "config:recommended"
    ],
    "packageRules": [
      {
        "matchCategories": [
          "docker"
        ],
        "fileMatch": [
          "(^|/)Earthfile$",
          "(^|/|\\.)Dockerfile$",
          "(^|/)Dockerfile[^/]*$"
        ]
      }
    ]
  }
}
```

I'm not sure what I did wrong here and would love to get a small hint in the right direction.

Thanks!

### Logs (if relevant)

https://developer.mend.io/github/fgreinacher/renovate-earthfile/-/job/8bc3265c-4b05-4b20-8df8-865e0ff3cc94 

From https://github.com/fgreinacher/renovate-earthfile

**Comment by rarkins** [comment, marked as answer]
The manager is called `dockerfile`, not `docker`, so change to this:

```
{
  "extends": ["config:base"],
  "dockerfile": {
    "fileMatch": [
      "(^|/)Earthfile$",
      "(^|/|\\.)Dockerfile$",
      "(^|/)Dockerfile[^/]*$"
    ]
  }
}
```

**Reply by fgreinacher** [reply]
I knew I must have made a stupid mistake, thanks a lot!

Would you mind editing the snippet in the comment https://github.com/renovatebot/renovate/issues/15975#issuecomment-1214640697 to prevent others running in the same issue? 🙇

**Reply by rarkins** [reply]
Unfortunately it can't be edited after being converted to a discussion
